lib/src/Base/Stat/LatentVariableModel.cxx

```diff
@@ -0,0 +1,299 @@
+//                                               -*- C++ -*-
+/**
+ *
+ *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/LatentVariableModel.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Exception.hxx"
+#include "openturns/SpecFunc.hxx"
+#include "openturns/Polygon.hxx"
+#include "openturns/Text.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(LatentVariableModel)
+
+static const Factory<LatentVariableModel> Factory_LatentVariableModel;
+
+/** Parameters constructor */
+LatentVariableModel::LatentVariableModel(const UnsignedInteger nLevels,
+										 const UnsignedInteger latentDim)
+  : CovarianceModelImplementation(Point(1, 1.0), Point(1, 1.0))
+  , latentDim_(latentDim)
+  , nLevels_(nLevels)
+  , latentCovarianceMatrix_(nLevels_)
+  , latentCovarianceModel_(latentDim_)
+{
+  if (latentDim_ < 1) throw InvalidArgumentException(HERE) << "Error: the dimension of the latent space must be >= 1";
+  if (nLevels_ < 2) throw InvalidArgumentException(HERE) << "Error: the number of discrete levels must be >= 2";
+  activeLatentCoordinateDim_ = 1 + latentDim_ * (nLevels_ - 2);
+  activeLatentVariables_ = Point(activeLatentCoordinateDim_, 0.0);
+  fullLatentVariables_ = Sample(nLevels_, latentDim_);
+  Indices activeParameter = Indices(inputDimension_ + outputDimension_ + activeLatentCoordinateDim_);
+  activeParameter.fill();
+  setActiveParameter(activeParameter);
+  updateLatentCovarianceMatrix();
+}
+
+/* Virtual constructor */
+LatentVariableModel * LatentVariableModel::clone() const
+{
+  return new LatentVariableModel(*this);
+}
+
+/* Computation of the covariance  function */
+Scalar LatentVariableModel::computeAsScalar(const Scalar z1, const Scalar z2) const
+{
+  /*checks z1 and z2 being among the known levels*/
+  const bool isLevelz1 = (z1 >= 0 && z1 < nLevels_ && (z1 - floor(z1)) == 0);
+  const bool isLevelz2 = (z2 >= 0 && z2 < nLevels_ && (z2 - floor(z2)) == 0);
+
+  if (!isLevelz1 || !isLevelz2) throw InvalidArgumentException(HERE) << "Error: the input discrete variables values: " << z1 
+  << ", and/or " << z2 << " are not among the known levels. They should both present integer values between 0 and l-1.";
```

Comment:
I guess that the "<<" continues the printing, but the indentation makes it confusing.
```suggestion
    << ", and/or " << z2 << " are not among the known levels. They should both present integer values between 0 and l-1.";
```